direct-query-core/src/main/java/org/opensearch/sql/directquery/rest/model/ExecuteDirectQueryRequest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.directquery.rest.model;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import org.opensearch.sql.directquery.model.DataSourceOptions;
+import org.opensearch.sql.prometheus.model.PrometheusOptions;
+import org.opensearch.sql.spark.rest.model.LangType;
+
+/*
+ * @opensearch.experimental
+ */
+@Data
+@NoArgsConstructor
+public class ExecuteDirectQueryRequest {
+  // Required fields
+  private String dataSources; // Required: From URI path parameter or request body
+  private String query; // Required: String for Prometheus, object for CloudWatch
+  @Setter private LangType language; // Required: SQL, PPL, or PROMQL
+  private String sourceVersion; // Required: API version
+
+  // Optional fields
+  private Integer maxResults; // Optional: limit for Prometheus, maxDataPoints for CW
+  private Integer timeout; // Optional: number of seconds
+  private DataSourceOptions options; // Optional: Source specific arguments
+  private String sessionId; // For session management
```

Comment:
i don't think we need CW comments here? also might be better to use javadoc syntax